diff --git a/include/linux/stddef.h b/include/linux/stddef.h
index e69de29..0000000 100644
--- a/include/linux/stddef.h
+++ b/include/linux/stddef.h
@@ -1,6 +1,9 @@
+#if __STDC_VERSION__ < 202000L
 enum {
     false   = 0,
     true    = 1
 };
+#endif

diff --git a/include/linux/types.h b/include/linux/types.h
index e69de29..0000000 100644
--- a/include/linux/types.h
+++ b/include/linux/types.h
@@ -1,2 +1,5 @@
+#if __STDC_VERSION__ < 202000L
 typedef _Bool                   bool;
+#endif
