--- a/natflow_common.c
+++ b/natflow_common.c
@@ -29,7 +29,19 @@
 static inline ip_set_id_t compat_ip_set_get_byname(struct net *net, const char *name, struct ip_set **set)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
+	struct {
+		struct nlattr nla;
+		char name[32];
+	} attr;
+	attr.nla.nla_len = sizeof(struct nlattr) + strlen(name) + 1;
+	attr.nla.nla_type = 2;
+	strncpy(attr.name, name, 32);
+	attr.name[31] = '\0';
+	return ip_set_get_byname(net, &attr.nla, set);
+#else
 	ip_set_id_t id = ip_set_id_find_byname(net, name);
 	if (id != IPSET_INVALID_ID)
 		ip_set_get_byindex(net, id);
 	return id;
+#endif
 }
